src/wp-includes/media.php

```diff
@@ -5564,3 +5571,288 @@ function wp_increase_content_media_count( $amount = 1 ) {
+	 * @param string $filtered_image     The filtered image. html including img tag
+	 * @param string $context            The context of the image.
+	 */
+	$check = apply_filters( 'img_tag_add_dominant_color', true, $attachment_id, $image_meta, $filtered_image, $context );
+	if ( ! $check ) {
+		return $filtered_image;
+	}
+
+	$data        = '';
+	$style       = '';
+	$extra_class = '';
+
+	if ( ! empty( $image_meta['dominant_color'] ) ) {
+		$data .= sprintf( 'data-dominant-color="%s" ', esc_attr( $image_meta['dominant_color'] ) );
+		$style = 'style="--dominant-color: #' . esc_attr( $image_meta['dominant_color'] ) . ';" ';
+	}
+
+	if ( isset( $image_meta['has_transparency'] ) ) {
+		$transparency = $image_meta['has_transparency'] ? 'true' : 'false';
+		$data        .= sprintf( 'data-has-transparency="%s" ', $transparency );
+		$extra_class  = $image_meta['has_transparency'] ? 'has-transparency' : 'not-transparent';
+	}
+
+	if ( ! empty( $data ) || ! empty( $style ) || ! str_contains( $style, $filtered_image ) || ! str_contains( $data, $filtered_image ) ) {
+		$filtered_image = str_replace( '<img ', '<img ' . $data . $style, $filtered_image );
+	}
+	if ( ! empty( $extra_class ) ) {
+		$filtered_image = str_replace( ' class="', ' class="' . $extra_class . ' ', $filtered_image );
+	}
+
+	return $filtered_image;
+}
+
+/**
+ * Add CSS needed for to show the dominant color as an image background.
+ *
+ * @since 6.1
+ */
+function dominant_color_add_inline_style() {
+	if ( ! current_theme_supports( 'dominant-color', 'inline-css' ) ) {
+		return;
+	}
+	$handle = 'dominant-color-styles';
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = 'img[data-dominant-color]:not(.has-transparency) { background-color: var(--dominant-color); background-clip: content-box, padding-box; }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_filter( 'wp_enqueue_scripts', 'dominant_color_add_inline_style' );
+
+/**
+ * Computes the Dominant color of the given attachment image.
+ *
+ * @since 6.1
+ * @access private
+ *
+ * @param int $attachment_id The attachment ID.
+ * @return array|WP_Error Array with the Dominant color and has transparency values or WP_Error on error.
+ */
+function _dominant_color_get_dominant_color_data( $attachment_id ) {
+	$mime_type = get_post_mime_type( $attachment_id );
+	if ( 'application/pdf' === $mime_type ) {
+		return new WP_Error( 'no_image_found', __( 'Unable to load image.' ) );
+	}
+	$file = wp_get_attachment_file_path( $attachment_id );
+	if ( ! $file ) {
+		$file = get_attached_file( $attachment_id );
+	}
+
+	$editor = wp_get_image_editor(
+		$file,
+		array(
+			'methods' => array(
+				'get_dominant_color',
+				'has_transparency',
+			),
+		)
+	);
+
+	if ( is_wp_error( $editor ) ) {
+		return $editor;
+	}
+
+	// can the image type be transparent if not lets shortcut here
+	if ( in_array( $mime_type, array( 'image/jpeg', 'image/bmp' ), true ) ) {
```

Comment:
@pbearne Why is this needed? There are a lots of mime types the imagick supports that do not support transpency. I would love to avoid this if we can.

Reply:
It saves a compute time.
I did put it in the editor, but it seems to me that we can save a few cycles and handle the images the same by doing it here